src/org/labkey/test/LabKeySiteWrapper.java

```diff
@@ -720,6 +722,31 @@ else if (getDriver().getTitle().startsWith("Sign In"))
                     // Note: leave the self-report setting unchanged
                     customizeSitePage.save();
                 }
+
+                /*
+                    Waiting for search service to boot up
+                    Issue 50601: PDF indexing is slow on first file after server startup on Windows
+                 */
+                Connection connection = createDefaultConnection();
+                SimpleGetCommand command = new SimpleGetCommand("search", "json");
+                command.setParameters(Map.of("q", "pinging to check server is started", "scope", "All"));
+                long searchEndTime = System.currentTimeMillis() + 180000; // Wait for maximum of 3 mins.
+                do
+                {
+                    try
+                    {
+                        command.execute(connection, "/");
```

Comment:
How about adding a `break;` here. If the command runs and there is no exception there is no need to continue.